### PR TITLE
fix(permissions): ignore empty values

### LINKED ISSUE #15445: Trailing comma in --allow-net causes panic
Including a trailing `,` in a `--allow-net` argument cases a panic.

```console
❯ RUST_BACKTRACE=1 deno run --allow-net=0.0.0.0:8080, src/mod.ts

============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos x86_64
Version: 1.24.2
Args: ["deno", "run", "--allow-net=0.0.0.0:8080,", "src/mod.ts"]

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: EmptyHost', runtime/permissions.rs:220:60
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: deno_runtime::permissions::Permissions::new_net
   4: deno_runtime::permissions::Permissions::from_options
   5: deno::run_command::{{closure}}
   6: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   7: deno::main::{{closure}}
   8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   9: deno_runtime::tokio_util::run_local
  10: deno::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

```

### PR DESCRIPTION
**Pull request by crowlKats**
Fixes #15445

### CODE REVIEW

File: runtime/permissions.rs [hunk 1]
```diff
@@ -1184,7 +1184,13 @@ impl Permissions {
         .as_ref()
         .map(|v| {
           v.iter()
-            .map(|x| NetDescriptor::from_string(x.clone()))
```
Comment (dsherret) [inline runtime/permissions.rs:1187]: I feel like instead we should change `NetDescriptor` to return a `Result<Self, ..>` then surface these errors to the user?

Comment (crowlKats) [reply runtime/permissions.rs:1187]: @dsherret done

File: cli/main.rs [hunk 2]
```diff
@@ -994,7 +994,8 @@ async fn run_with_watch(flags: Flags, script: String) -> Result<i32, AnyError> {
     ModuleSpecifier,
   )| {
     let flags = flags.clone();
-    let permissions = Permissions::from_options(&flags.permissions_options());
+    let permissions =
+      Permissions::from_options(&flags.permissions_options()).unwrap();
```
Comment (crowlKats) [inline cli/main.rs:998]: this shouldnt be unwrapped

**Review by curtisallen — APPROVED** [review]
LGTM thanks!

**Review by dsherret — APPROVED** [review]
Looks great!

**Pull request merged**